mono/profiler/proflog.c

```diff
@@ -2027,16 +2027,20 @@ mono_sample_hit (MonoProfiler *profiler, unsigned char *ip, void *context)
 	InterlockedIncrement (&sample_hits);
 
 	uint64_t now = current_time ();
+	SampleHit *sample;
 
-	SampleHit *sample = (SampleHit *) mono_lock_free_queue_dequeue (&profiler->sample_reuse_queue);
-	gboolean hazardous;
+again:
+	sample = (SampleHit *) mono_lock_free_queue_dequeue (&profiler->sample_reuse_queue);
 
-	if (sample && (hazardous = mono_thread_is_pointer_hazardous (sample))) {
+	if (sample && mono_thread_is_pointer_hazardous (sample)) {
```

Comment:
This looks wrong. Why would a a sample from the reuse queue be hazardous in the first place?